### ISSUE TITLE
Include parsed Query String parameters from a GET request (part in the URL after question mark) in q.args

### ISSUE THREAD

**Issue by dirkdevriendt** [h2oai/wave#500]
#### Is your feature request related to a problem? Please describe

- Ability to provide a unique link (get request, QR code etc) to users, so they arrive directly on a personalised page
- Didn't really check this, but it seems wave does not recognise the semantic meaning of the `?` in a URL

#### Describe the solution you'd like

Have access in the app to either a parsed list of URL parameters or the complete location/URL as a whole, so it can be parsed in the app itself.

#### Describe alternatives you've considered

Currently the only way is to either:
- use a # with a unique id
- (ab)use the # by also adding a ? in the URL, grab the # arg and parse it, which I assume messes with waves "standard" hash routing

**Comment by lo5** [comment]
@dirkdevriendt Correct. Currently the only way is to use `#foo/bar/etc`. Doing this does not affect/break anything in Wave - the framework doesn't do anything special with the '#' slugs other than hand it over to the app using `q.args['#']`, if available.

That said, we can add the ability for the front-end to pass query string params as well - that would definitely be useful like you indicated - thanks for the suggestion!
